domain/externalcontroller/doc.go

```diff
@@ -0,0 +1,6 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package externalcontroller provides a service to keep track of external
```

Comment:
I agree, it does sound a bit weird. I started the sentence with package because it seems to be the done thing for Go packages. The styleguide [here](https://google.github.io/styleguide/go/decisions#package-comments) does not explicitly say it but its the way all the main go packages seem to do it.

I used "related to" because there is a juju relation between the application.

On reflection I've removed the second half of the sentence to keep it more general. I don't think I know enough about this domain to make a much better doc. But at least its there now so people can add to it.

Reply:
Yes, sorry the scope of my suggestion was not the `// Package xx` part (even if my comment is misguiding) which is, as you say, a convention in Go (for the sake of Godoc automation, AMHO).

The weird feeling was more about the second part. Removing it is fine. If we want to have more explanation we should add a paragraph, because we are dealing with how juju concept interacts with each other, and it cannot be done in one sentence without being very unconfortable to understand :D (or we would have a lot less work to do on documenation :D)